src/content/docs/developer-guide/git-guide.md

```diff
@@ -0,0 +1,178 @@
+
+With git, you can use any username and email address you want to make a commit, even a name and email that does not belong to you! In small teams, or closed projects, this is not an issue. However, in open source work, it can create confusion.
+
+Thankfully, there are several ways to sign your commits and confirm your identity. We require commits to be signed for our repo, because it helps us improve the security of the project.
+
+To find out more about how to sign your commits, check out the [official GitHub documentation on the topic](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
+
+## Skipping the pre-commit hooks
+
+We use pre-commit and commit hooks to run linting, tests, and other types of checks. To skip these tests, you can use the `--no-verify` flag, or its shorthand `-n`.
+
+```sh
+git commit --no-verify -m “message”
+```
+
+The `--no-verify` flag is also available for the `git push` command, and it serves the same function. 
+
+However, it's important to keep in mind that these tests are only skipped locally, and the flag is generally used to speed up development. Once you make a PR, you have to ensure that all the tests complete successfully.
+
+## Good practices
+
+### git rebase
+
+As you contribute to open source projects, you may run into conflicts between your PR or branch and the upstream code. This usually happens when your commit is made on an older version of the code base, and new changes have been implemented by other contributors.
+
+With the `git rebase` command you can move your branch along the history of the repo and change the commit it is based on. This will allow you to base your branch on the latest commit. The command for this is:
+
+```sh
+git rebase origin/main
+```
+
+It's important to note that `origin/main` has to be up to date with `ratatui/main`. This can be done in a number of ways, including by [setting `ratatui/main` as the upstream for your repo](#upstream), or through the GitHub interface.
+
+![Updating a branch on Github](./update.png)
+
+Rebasing is very straightforward in theory. However, you may run into conflicts that have to be resolved before you can rebase your branch. One of the main causes for rebase conflicts is the presence of many small commits in your branch history. This is where squashing commits is useful.
+
+To squash commits, you will have to run an interactive rebase using the `-i` flag.
+
+Make sure that you are on the right branch first, using:
+
+```sh
+git checkout your-branch
+```
+
+And then:
+
+```sh
+git rebase -i HEAD~x
+```
+
+In this scenario, `HEAD` refers to the latest commit of the main branch, and `x` represents the number of commits that are on the branch. To find out what this number is you will need to run:
+
+```sh
+git log --branches[=<branch>]
+```
+
+The `--branches` flag is useful when you’re working on a project that has a lot of commits. It allows you to check only the commits made on your branch.
```

Comment:
I didn't know about this flag, I had a hard time using it, and even then this isn't clear for a new user how many commits are in his branch.
What about `git log main..HEAD`, `main` could also be `origin/main` or `upstream/main`, not sure which is the best here.